pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -390,10 +419,33 @@ func addColumn(b BuildCtx, spec addColumnSpec) (backing *scpb.PrimaryIndex) {
 		updatedPartitioning.IndexID = temp.IndexID
 		b.Add(updatedPartitioning)
 	}
-
+	for _, ec := range append(existingColumns, ic) {
+		cloned := protoutil.Clone(ec).(*scpb.IndexColumn)
+		cloned.IndexID = temp.IndexID
+		b.Add(cloned)
+	}
 	return replacement
 }
 
+func getNextStoredIndexColumnOrdinal(
+	allTargets ElementResultSet, idx *scpb.PrimaryIndex,
+) (ord uint32) {
+	var foundAny bool
```

Comment:
nit: just initialize return `ord = -1` we can get rid of `foundAny` and just `return ord ++`